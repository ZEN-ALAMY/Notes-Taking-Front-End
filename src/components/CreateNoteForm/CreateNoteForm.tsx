import React, { useState } from "react";
import { addNote } from "../../services/notes-taking.service";

function Input() {
  const [values, setValues] = useState({
    title: "",
    date: "",
    relevance: "",
  });
  const titleHandler = (event) => {
    setValues({ ...values, title: event.target.value });
  };

  const submitHandler = async (event) => {
    event.preventDefault();

    console.log(values);
    const response = await addNote(values);
    // const responseData = await response.json();
    setValues({ ...values, title: "", date: "", relevance: "" });
  };

  return (
    <div className="xl:w-[650px] sm:w-[360px] h-[480px]  rounded-2xl mt-12 mx-auto">
      <form className="flex flex-col">
        <input
          className="w-[300px] xl:w-[500px] sm:w-[400px] h-[70px] mx-auto my-[20px] rounded-lg border-[3px] border-black border-solid mt-[40px]"
          type="text"
          required
          value={values.title}
          placeholder="NOTES TITLE"
          name="firstName"
          onChange={titleHandler}
        />

        <input
          className="w-[300px] xl:w-[500px] sm:w-[400px] h-[70px] mx-auto my-[20px] rounded-lg border-[3px] border-black border-solid"
          type="text"
          value={values.date}
          placeholder="DATE"
          name="date"
          onChange={(event) => {
            setValues({ ...values, date: event.target.value });
          }}
        />
        <input
          className="w-[300px] xl:w-[500px] sm:w-[400px] h-[70px]  mx-auto my-[20px] rounded-lg border-[3px] border-black border-solid"
          type="text"
          value={values.relevance}
          name="relevance"
          onChange={(event) => {
            setValues({ ...values, relevance: event.target.value });
          }}
        />
        <button
          className="w-[200px] mx-auto mb-[30px] mt-[10px] h-[60px] bg-green-500 sm:bg-red-500 xl:bg-blue-500 text-black rounded-xl font-extrabold text-lg"
          type="submit"
          onClick={submitHandler}
        >
          SUBMIT
        </button>
      </form>
    </div>
  );
}

export default Input;
