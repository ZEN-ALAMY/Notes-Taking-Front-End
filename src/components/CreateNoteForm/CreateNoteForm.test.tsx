import { ChangeEvent } from "react";
import { fireEvent, render, waitFor } from "@testing-library/react";
import CreateNoteForm from "../CreateNoteForm/CreateNoteForm";

describe("Create note form tests", () => {
  it("should render the UI correctly", () => {
    expect(render(<CreateNoteForm />).asFragment()).toMatchSnapshot();
  });

  it("should change the values correctly", async () => {
    const { getByPlaceholderText } = render(<CreateNoteForm />);

    const titleInput = getByPlaceholderText("NOTES TITLE") as HTMLInputElement;

    await waitFor(() => {
      fireEvent.change(titleInput, { target: { value: "Test Note" } });
      expect(titleInput.value).toBe("Test Note");
    });
  });

  it("should submit values correctly", async () => {
    const { container } = render(<CreateNoteForm />);

    const relevanceInput = container.querySelector(
      `[name = 'relevance']`
    ) as HTMLInputElement;

    if (relevanceInput) {
      await waitFor(() => {
        fireEvent.change(relevanceInput, { target: { value: "Test Note" } });
        expect(relevanceInput.value).toBe("Test Note");
      });
    }
  });
});
