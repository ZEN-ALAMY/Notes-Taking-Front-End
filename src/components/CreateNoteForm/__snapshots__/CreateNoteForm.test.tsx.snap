// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create note form tests should render the UI correctly 1`] = `
<DocumentFragment>
  <div
    class="xl:w-[650px] sm:w-[360px] h-[480px]  rounded-2xl mt-12 mx-auto"
  >
    <form
      class="flex flex-col"
    >
      <input
        class="w-[300px] xl:w-[500px] sm:w-[400px] h-[70px] mx-auto my-[20px] rounded-lg border-[3px] border-black border-solid mt-[40px]"
        name="firstName"
        placeholder="NOTES TITLE"
        required=""
        type="text"
        value=""
      />
      <input
        class="w-[300px] xl:w-[500px] sm:w-[400px] h-[70px] mx-auto my-[20px] rounded-lg border-[3px] border-black border-solid"
        name="date"
        placeholder="DATE"
        type="text"
        value=""
      />
      <input
        class="w-[300px] xl:w-[500px] sm:w-[400px] h-[70px]  mx-auto my-[20px] rounded-lg border-[3px] border-black border-solid"
        name="relevance"
        type="text"
        value=""
      />
      <button
        class="w-[200px] mx-auto mb-[30px] mt-[10px] h-[60px] bg-green-500 sm:bg-red-500 xl:bg-blue-500 text-black rounded-xl font-extrabold text-lg"
        type="submit"
      >
        SUBMIT
      </button>
    </form>
  </div>
</DocumentFragment>
`;
